src/develop/develop.c

```diff
@@ -141,9 +141,8 @@ void dt_dev_init(dt_develop_t *dev, int32_t gui_attached)
     // (regardless of ppd). Try to get enough detail for a (default)
     // 350px panel, possibly 2x that on hidpi.
     dev->histogram_waveform_width = darktable.mipmap_cache->max_width[DT_MIPMAP_F]/2;
-    // Hardcoded hack: histogram widget will probably be either 175 or
-    // 350 pixels high depending on hidpi
-    dev->histogram_waveform_height = 175;
+    const float histheight = dt_conf_get_int("histogram_height") * 1.0f;
+    dev->histogram_waveform_height = DT_PIXEL_APPLY_DPI(histheight);
```

Comment:
I don't think you actually need to change the `histogram_waveform_height` to change the histogram widget size. Cairo draws the histogram buffer (`dev->histogram_waveform`) scaled to the width/height of the histogram widget, and should look OK even if it's not in a 1:1 (or 1:2 for hidpi) ratio to the widget.

It may be worth only adjusting the widget and not adjusting the buffer -- or having a separate config for the buffer size. This is because the memory use and processing time of calculating the waveform increase in proportion to `dev->histogram_waveform_height` even though the image quality doesn't improve that much.

Reply:
I think you're probably right here, but the previous height was 175 and its max size is limited to 200. I suspect most people will _reduce_ the height (or that's the impression I get from screenshots people have posted on other issues) which will reduce the processing time and memory use. Please note that a subsequent PR removed the use of DPI_PIXEL_APPLY_DPI, so the waveform height is now limited to between 100 and 200.

Feel free to change it if you end up submitting a PR for #3005.